spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaAdminProperties.java

```diff
@@ -26,31 +25,13 @@
  * @author Gary Russell
  * @since 2.0
  *
+ * @deprecated in favor of {@link KafkaTopicProperties}
  */
-public class KafkaAdminProperties {
-
-	private Short replicationFactor;
-
-	private Map<Integer, List<Integer>> replicasAssignments = new HashMap<>();
+@Deprecated
+public class KafkaAdminProperties extends KafkaTopicProperties {
 
 	private Map<String, String> configuration = new HashMap<>();
 
-	public Short getReplicationFactor() {
-		return this.replicationFactor;
-	}
-
-	public void setReplicationFactor(Short replicationFactor) {
-		this.replicationFactor = replicationFactor;
-	}
-
-	public Map<Integer, List<Integer>> getReplicasAssignments() {
-		return this.replicasAssignments;
-	}
-
-	public void setReplicasAssignments(Map<Integer, List<Integer>> replicasAssignments) {
-		this.replicasAssignments = replicasAssignments;
-	}
-
 	public Map<String, String> getConfiguration() {
 		return this.configuration;
```

Comment:
No need for a local map; delegate to `getProperties()` and `setProperties(configuration)`.